Framework/Algorithms/src/FindPeaksConvolve.cpp

```diff
@@ -0,0 +1,376 @@
+  }
+
+  return err_str;
+}
+
+void FindPeaksConvolve::storeClassProperties() {
+  m_inputDataWS = getProperty("InputWorkspace");
+  m_createIntermediateWorkspaces = getProperty("CreateIntermediateWorkspaces");
+  m_findHighestDatapointInPeak = getProperty("FindHighestDataPointInPeak");
+  m_iOverSigmaThreshold = getProperty("IOversigmaThreshold");
+  m_performBinaryClosing = getProperty("PerformBinaryClosing");
+
+  int wsIndex = getProperty("WorkspaceIndex");
+  if (wsIndex == EMPTY_INT()) {
+    m_specCount = static_cast<int>(m_inputDataWS->getNumberHistograms());
+    m_specNums.resize(m_specCount);
+    std::iota(std::begin(m_specNums), std::end(m_specNums), 0);
+  } else {
+    m_specCount = 1;
+    m_specNums.push_back(wsIndex);
+  }
+  m_peakResults.resize(m_specCount);
+}
+
+void FindPeaksConvolve::performConvolution(const size_t dataIndex) {
+  const auto specNum{m_specNums[dataIndex]};
+  const HistogramData::HistogramX *xData{&m_inputDataWS->x(specNum)};
+  const auto kernelBinCount{getKernelBinCount(xData)};
+  const Tensor1D kernel{createKernel(static_cast<int>(kernelBinCount))};
+
+  const auto binCount{m_inputDataWS->getNumberBins(specNum)};
+  const double paddingSize = (std::ceil(static_cast<double>(kernelBinCount) * 1.5) - 2) / 2;
+  const TensorMap_const yData{&m_inputDataWS->y(specNum).front(), binCount};
+  Eigen::array<std::pair<double, double>, 1> paddings{std::make_pair(std::ceil(paddingSize), std::floor(paddingSize))};
+  const auto yData_padded{yData.pad(paddings)};
+  const Eigen::array<ptrdiff_t, 1> dims({0});
+  const Tensor1D yConvOutput{yData_padded.convolve(kernel, dims)};
+
+  const auto eData{TensorMap_const{&m_inputDataWS->e(specNum).front(), binCount}.pad(paddings)};
+  const Tensor1D eConvOutput{eData.square().convolve(kernel.square(), dims).sqrt()};
+
+  const Tensor1D smoothKernel{
+      createSmoothKernel(static_cast<size_t>(std::ceil(static_cast<double>(kernelBinCount) / 2.0)))};
+  const Tensor1D iOverSigConvOutput{(yConvOutput / eConvOutput).convolve(smoothKernel, dims)};
+
+  std::unique_ptr<EigenMap_const> xDataPostConv;
+  Eigen::VectorXd xDataCentredBins;
+  if (xData->size() != static_cast<size_t>(yData.size())) {
+    xDataCentredBins = centreBinsXData(xData);
+    xDataPostConv = std::make_unique<EigenMap_const>(EigenMap_const{xDataCentredBins.data(), xDataCentredBins.size()});
+  } else {
+    xDataPostConv = std::make_unique<EigenMap_const>(EigenMap_const(&xData->front(), xData->size()));
+  }
+  extractPeaks(dataIndex, iOverSigConvOutput, *xDataPostConv, yData, kernelBinCount / 2);
+
+  if (m_createIntermediateWorkspaces) {
+    createIntermediateWorkspaces(dataIndex, kernel, iOverSigConvOutput, *xDataPostConv);
+  }
+}
+
+Tensor1D FindPeaksConvolve::createKernel(const int binCount) {
+  Tensor1D kernel(binCount);
+  for (int i{0}; i < binCount; i++) {
+    // consider if these 0.25/0.75 values should be inputs
```

Comment:
I think for now they should not be inputs yet - maybe this comment should be replaced with something explanatory e.g. 
```
create an integrating shoebox kernel with a central positive region and negative background shell with the same (or as close as possible) number of elements in each
```
The reason being we are subtracting a background from a signal and we want to make sure we have similar stats in each. The only reason to make the shell smaller would be if you knew you had lots of overlapping peaks, but it seems to do a good job even on overlapping peaks!